python/datasources/vera_incarceration_county.py

```diff
@@ -474,9 +498,55 @@ def generate_partial_breakdown(df, demo_type, data_type, property_type):
         columns=col_rename_map)
 
     # melt into HET style df with a row per GEO/DEMO combo
-    df = df.melt(id_vars=GEO_COLS_TO_STANDARD.values(),
+    df = df.melt(id_vars=[std_col.TIME_PERIOD_COL, *GEO_COLS_TO_STANDARD.values()],
                  value_vars=value_vars,
                  var_name=het_group_column,
                  value_name=het_value_column)
 
     return df
+
+
+def add_confined_children_col(df):
+    """ Parameters: df: pandas df containing the entire Vera csv file.
+    Returns same df replacing juvenile cols with a summed, rounded `total_confined_children` col
+    """
+    df[CHILDREN] = df[JUVENILE_COLS].sum(
+        axis="columns", numeric_only=True).round(0)
+    df = df.drop(columns=JUVENILE_COLS)
+
+    return df
+
+
+def add_jail_pct_share_col(df, demo_type: SEX_RACE_ETH_AGE_TYPE):
```

Comment:
Is there a good reason to not massage all of the data to be ready for our util functions and then run those on them? If the main issue is that the sum wont add up to the total, maybe write a wrapper around the util function that accounts for that case (but leave it in this file so people dont accidentally use it)

Reply:
Age didn't change because we don't have age groups anyways, but for sex and race I basically overwrite the ALL raw_count with the SUM OF GROUPS raw_count, and then proceed as normal and calc the pct_share after the melt using the normal util function. Saves having to keep track of the two types of "ALL" rows and we don't need the raw_count for per_100k calcs since vera provides those